pkg/agent/controller/endpoint.go

```diff
@@ -274,7 +275,7 @@ func allAddressesIPv6(addresses []corev1.EndpointAddress) bool {
 }
 
 func (e *EndpointController) getIP(address *corev1.EndpointAddress) string {
-	if e.isHeadless && e.globalIngressIPCache != nil {
+	if e.clusterID != e.localClusterID && e.isHeadless && e.globalIngressIPCache != nil {
```

Comment:
This will work for local cluster but break remote clusters. This is part of code where we create EndpointSlices from service's endpoints to sync to remote clusters. ClusterId check here is irrelevant as we will only hit this code for local services on lighthouse agent.

The files you need to modify and handle clusterIDs are in pkg/endpointslice/